### PR TITLE
Making optionalZipCountries JS code conditional

### PR DESCRIPTION
**Pull request by loekvangool**
`js/varien/form.js` checks for existence of the variable, hence it's allowed to be undefined if empty.

There is more on this. The block is inserted into every Magento page, which seems excessive. It's probably only used in the Cart but I haven't checked thoroughly so leaving it alone for now. Same for the admin.

Also `type="text/javascript"` is no longer necessary.

### CODE REVIEW

**Review by luigifab — COMMENTED** [review]
Why not, can you fix indent, use `endif` in template, and change `Mage::helper` with `$this->helper`? (not yet tested)

**Comment by loekvangool** [comment]
I will never understand why templates are written in this horrible coding standard, lol.

File: app/design/frontend/base/default/template/directory/js/optional_zip_countries.phtml [hunk 1]
```diff
@@ -26,8 +26,13 @@
...
 ?>
-<script type="text/javascript">
-//<![CDATA[
-optionalZipCountries = <?php echo $this->helper('directory')->getCountriesWithOptionalZip(true) ?>;
-//]]>
-</script>
+<?php
+$optional_zip_countries = $this->helper('directory')->getCountriesWithOptionalZip(false);
+if(!empty($optional_zip_countries)):
+  ?>
```
[the remark this answers is not in the document]
Comment (ADDISON74) [reply app/design/frontend/base/default/template/directory/js/optional_zip_countries.phtml:32]: It looks better.

**Comment by kiatng** [comment]
@sreichel installed phpmd and phpcs:
image

One of the rules is camelCase. Not against snake_case, but it is better to standardize on camelCase for everything?

File: app/design/frontend/base/default/template/directory/js/optional_zip_countries.phtml [hunk 2]
```diff
@@ -26,8 +26,11 @@
...
-<script type="text/javascript">
-//<![CDATA[
-optionalZipCountries = <?php echo $this->helper('directory')->getCountriesWithOptionalZip(true) ?>;
-//]]>
-</script>
+<?php $optional_zip_countries = $this->helper('directory')->getCountriesWithOptionalZip(false) ?>
+<?php if(!empty($optional_zip_countries)): ?>
+  <script>
+  //<![CDATA[
+  optionalZipCountries = <?php echo $this->helper('core')->jsonEncode($optional_zip_countries); ?>;
```
Comment (fballiano) [inline app/design/frontend/base/default/template/directory/js/optional_zip_countries.phtml:33]:
```suggestion
<?php $optionalZipCountries = $this->helper('directory')->getCountriesWithOptionalZip(false) ?>
<?php if(!empty($optionalZipCountries)): ?>
  <script>
  //<![CDATA[
  optionalZipCountries = <?php echo $this->helper('core')->jsonEncode($optionalZipCountries); ?>;
```

Comment (fballiano) [reply app/design/frontend/base/default/template/directory/js/optional_zip_countries.phtml:33]: I don't love camelcase for variables but @kiatng is right

**Comment by loekvangool** [comment]
good to go

**Review by fballiano — APPROVED** [review]

**Review by ADDISON74 — APPROVED** [review]

**Comment by fballiano** [comment]
merged and 20ed

**Pull request merged**